### DISCUSSION TITLE
Forma de actualizar un schemaInstance de una property?

### DISCUSSION THREAD

**Discussion by NicolasPLG** [Cotalker/documentation#166] [2. Questions and Answers] [answered]
Resulta que quier actualizar los extras de una task con asset unico.

Como rutina tengo esto:

```
{
                            "_id" : ObjectId("609999c23e26400009787428"), 
                            "key" : "get_patch_tipo_de_solucion", 
                            "name" : "NWRequest", 
                            "data" : {
                                "url" : "$JOIN#/#($ENV#BASEURL)#api#v2#properties#jsonpatch#($OUTPUT#create_task_project#task|asset)", 
                                "method" : "PATCH", 
                                "headers" : {
                                    "authorization" : "$OUTPUT#get_auth#data|extra|auth", 
                                    "admin" : true
                                }, 
                                "body" : [
                                    {
                                        "op" : "replace", 
                                        "path" : "/schemaInstance/tipo_de_solucion", 
                                        "value" : "$OUTPUT#get_answers#data|data|[find=>identifier=ingrese_tipo_de_solucion]|responses|0|[json=>parse]|name|code"
                                    }
                                ]
                            }, 
                            "next" : {
                                "SUCCESS" : "get_patch_horas_proyecto", 
                                "ERROR" : ""
                            }, 
                            "customNetworkRequest" : [

                            ]
                        },
```
Pero por lo que veo a simple vista, creo que tengo un problema con el patch que lo tengo mal escrito.

A modo de imagen quiero actualizar los campos (ejemplo del nwr adjuntado:
![image](https://user-images.githubusercontent.com/61419471/117723161-0ae3e500-b1b0-11eb-9bbe-1b03416e35af.png)

Gracias

PD: si fuera por extratype sería fácil, porque lo he hecho, pero para los extras 3.0 (o los schemas), no he visto una actualización estos en las rutinas, y sería la primera implementación.

**Comment by rafaelwalter** [comment, marked as answer]
@agallardol podrias explicar a @NicolasPLG como actualizar las extensions

**Reply by acedward** [reply]
Se ve bien ese NWRequest. 
@NicolasPLG podrías enviar lo que apace en Logs cuando falla este bot

**Reply by NicolasPLG** [reply]
@acedward en los logs del navegador no aparece anda, en los automatic log aparece:
``` {
  "message": "STATUS CODE: 401"
}```

**Comment by agallardol** [comment]
@NicolasPLG a priori sería bueno saber si esta property es "antigua", de ser así, seguramente dicha property no tiene el campo property.schemaInstance y se hace válida toda la explicación que viene a continuación. De igual forma resulta muy útil para estos casos que nos compartas los logs de cuando falla este bot.

## Es una property antigua?

**Qué ocurre con esto? Explicación técnica**
Como estamos intentando reemplazar un valor en el campo property.schemaInstance.tipo_de_solucion y schemaInstance no existe, el patch, que por detrás está utilizando JSON Patch (por eso enviamos un conjunto de operaciones y valores) falla al intentar reemplazar el valor de una propiedad que no existe.

**Qué podemos hacer para corregirlo?**
- Dependiendo de las reglas de negocio del esquema de esa propiedad, podemos hacer una operación add sobre la property para añadir el campo que falta property.schemaInstance y luego ejecutar la operación que estás intentando hacer.
- Otro workaround posible es ir al admin y editar cualquier campo de esta property, esto hará que el campo property.schemaInstance se genere automáticamente.

**Me ocurrirá con otras properties?**
Esto no debería ocurrir con properties creadas recientemente, sí con properties antiguas.

@SeIturra Puede ayudar mucho con este tema, el fue de las primeras personas en levantar esta discusión.

**Reply by NicolasPLG** [reply]
gracias @agallardol por la información. Puede que el problema que estoy cometiendo es que se las estoy agregando al assets, en el momento de crear la tasks (1.- creo la task y luego quiero agregar los schemas. probé con add anteriormente y no me funcionó.

Sé como hacerlo cuando es un extratype 2.0 (la versión antigua de los schemas), pero aqui no. Si @SeIturra  tiene más información me gustaría que la colocara aqui

:>

**Reply by seiturra** [reply]
@NicolasPLG la etapa se ve bien. Lo único que hay que ver es que el campo "/schemaInstance/tipo_de_solucion" esté creado al crear el asset. Si no se crea automáticamente, te recomiendo editar el asset justo después de crear la tarea para agregarle el schemaInstance/tipo_de_solucion con un texto vacío. Algo así:

```
            {
                "key" : "create_schemainstance", 
                "name" : "NWRequest", 
                "data" : {
                    "url" : "$JOIN#/#($ENV#BASEURL)#api#properties#($OUTPUT#create_task#task|asset)", 
                    "method" : "PATCH", 
                    "defaultAuth" : true, 
                    "body" : {
                        "schemaInstance" : {
                            "tipo_de_solucion" : ""
                        }
                    }
                }, 
                "next" : {
                    "SUCCESS" : "", 
                    "ERROR" : ""
                }, 
                "customNetworkRequest" : [

                ]
            }, 
```
